### ISSUE TITLE
GameGuru MAX - Request - Crawl Added to Player

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#3035]
Currently, we have Crouch, but it's not low enough if we want the player to traverse a vent system, for example, where we want to simulate them crawling or doing a bit more than crouching down a bit. Would it be possible to get something closer to a crawl added or be able to adjust crouch if we need it to be even lower?

**Comment by Mt-Chiliad-UFO** [comment]
:~)  :~)  :~) Yeah, already doing this time ago in Classic. No problem, works well. Just ask the right person, and if they imagine really hard that diabolical MAX is fit to benefit from their efforts, they might donate that script to the crooked winds.

![Screenshot (998) smlpng](https://user-images.githubusercontent.com/99560780/166317811-8686a8f5-f044-46b1-8614-310e9f0ec16a.png)

**Comment by MonkeyFrogStudio** [comment]
According to AmenMoses, this is not easily done in MAX, which is why I am requesting it here. :)

**Comment by Mt-Chiliad-UFO** [comment]
Yeah, who knows...? They said the same with Classic though. No point in much testing in MAX or even opening that scam software. Perhaps I could give a try-- but it might take a couple hours for MAX to start and open up ready to test a script. 50x more hurdles to do scripting in MAX....
![Screenshot (999) cropped](https://user-images.githubusercontent.com/99560780/166321278-e16db814-9160-4b10-8fac-9f7e0741d8cc.png)

**Comment by Mt-Chiliad-UFO** [comment]
Works fine in MAX. No problem with it. AND.... just proved that any dynamic entities are BROKEN because the barrel that is running the script HAS NO COLLISION! :~) :~) :~)
Everything (including the bushes) has collision!

![Screenshot (1001) cropped](https://user-images.githubusercontent.com/99560780/166324437-c00d9db0-724b-4d39-a67d-c253c8141d10.png)

**Comment by MonkeyFrogStudio** [comment]
Okay, I am not sure what you're trying to prove here. I cannot, from a screen shot, see that the barrel has not collision. If there is a problem with dynamic physics, then report that in your own post here on GitHub. That's what GitHub is for. Please don't SPAM my post with this stuff. I had talked with AmenMoses about crawling, he stated it was more difficult due to the collision capsule being used, etc., so I made a request. Simple. I am no programmer. I didn't ask for any proof for anything else. So, not sure why you're bringing all this stuff up here to me. I didn't ask for proof about collision and dynamic entities.

But now that you've brought it up, did you turn your physics to ON for that barrel? Because, for me, I can collide with physics objects just fine.

**Comment by synchromesh62** [comment]
> Currently, we have Crouch, but it's not low enough if we want the player to traverse a vent system, for example, where we want to simulate them crawling or doing a bit more than crouching down a bit. Would it be possible to get something closer to a crawl added or be able to adjust crouch if we need it to be even lower?

I reported this some time back when creating my Trapped Demo.
The tunnel behind the picture in the demo and the height below the fireplace has to be around 4ft, 6 inches to clear which i agree is far to to high. 
In fact my fireplace is the limit i can crawl under which is just below the height of the Zombie In fact i redesigned my level to incorporate the fireplace as my original idea for a crawlspace did not work. 

![height](https://user-images.githubusercontent.com/35708217/166389280-7279cc88-4472-4825-bbe0-37c415b106a9.png)

**Comment by AmenMoses** [comment]
There is a Lua command to control the player capsule but it is limited:
ControlDynamicCharacterController(iObj, -- controls the physics capsule of the player
 fAngleY
 fAngleX
 fSpeed
 fJump
 fDucking
 fPushAngle
 fPushForce
 fThrustUpwards )

The fDucking I believe is simply a on/off flag, in order to have proper crawl control would need additional control over capsule height and orientation.

Of course this would then lead to all sorts of other issues, like how does the crawl get triggered, same button with multiple presses?  Different button?  

For full VR of course the player should mimic what the RL player is doing, driven by the height and orientation of the headset!

**Comment by MonkeyFrogStudio** [comment]
I think what we need is a special instance for "crawl spaces". Like they use in games like Alien Isolation. When you come to an area you can crawl in, like a vent or other crawl space, you get a prompt (press "E" to enter the crawl space) and that moves the camera/player into that space until they exit. It's a lot like getting onto a ladder and exiting a ladder. But this way, you don't need an extra key to control it. You just use whatever the action key is (like "E").

**Comment by Necrym59** [comment]
Can a command to rotate the capsule to a lying down position be done with the fAngleX be used? ive never looked at these commands.

**Comment by MonkeyFrogStudio** [comment]
@Mt-Chiliad-UFO I have no idea what your problem is. I didn't mention a particular "ladder script" at all. It's how ladders, in ANY game work. You just get on them. I play games and this is how they work in most instances (though not all). Why the heck are you jumping on me? What did I do to you? Am I the one that used your script? Remember, I'm not a programmer. And what does any of that have to do with my request here? Can you please just leave this thread, which is about a request for "crawl" to be added to MAX?

**Comment by Necrym59** [comment]
Ummm I dont know what his problem is but dont attempt to cast falshoods and outright lies on another people.

**Comment by synchromesh62** [comment]
Had a clean-up Guys and just left main the issue at hand.
The Github Its not a forum.

**Comment by MonkeyFrogStudio** [comment]
I really want to emphasize that it's pretty important in FPS games to have the ability to crawl through tight spaces, like vents. It's a staple of the genre. The fact that MAX cannot accommodate this is ... strange ... to say the least, imo. Please add the ability for the player to crawl in spaces like vents (where the end-user designates that crawling should take place).
